src/config/presets-loader.ts

```diff
@@ -0,0 +1,71 @@
+import path from 'path';
+import fse from 'fs-extra';
+
+import { Config } from './config-types';
+import { parseFile } from '@stackbit/utils';
+
+export class ConfigPresetsError extends Error {
+}
+
+export interface PresetsLoaderResult {
+    config: Config | null;
+    errors: ConfigPresetsError[];
+}
+
+export async function loadPresets(dirPath: string, config: Config): Promise<PresetsLoaderResult> {
+    const presetFiles = [];
+    const presetsRelDirs = ['.stackbit/presets', 'node_modules/@stackbit/components/presets'];
+
+    for (const presetsRelDir of presetsRelDirs) {
+        const presetsDir = path.join(dirPath, presetsRelDir);
+        if (!(await fse.pathExists(presetsDir))) {
+            continue;
+        }
+        presetFiles.push(...(await fse.readdir(presetsDir))
+            .filter(fileName => path.parse(fileName).ext === '.json')
+            .map(fileName => path.join(presetsRelDir, fileName))
+        );
+    }
+
+    const presets: any = {};
+    const presetsByModel: any = {};
+
+    for (const presetFile of presetFiles) {
+        const presetsRelDir = path.dirname(presetFile);
+        const presetPath = path.join(dirPath, presetFile);
+        const presetData = await parseFile(presetPath);
+        for (const [i, preset] of presetData.presets.entries()) {
```

Comment:
don't assume `presetData.presets` will be an array or that it will even exists, use:
```
_.forEach(_.get(presetData, 'presets', {}), (preset, i) => { ... });
```
or any other defensive coding

later we can return errors, but the code should not throw.